libraries/engage/page/widgets/NestedCategoryFilter/NestedCategoryFilter.jsx

```diff
@@ -0,0 +1,155 @@
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { I18n, ButtonLink } from '@shopgate/engage/components';
+import { bin2hex } from '@shopgate/pwa-common/helpers/data';
+import { CATEGORY_PATH } from '@shopgate/pwa-common-commerce/category/constants';
+import { makeStyles } from '@shopgate/engage/styles';
+import { themeConfig } from '@shopgate/pwa-common/helpers/config';
+import { useRoute, usePrevious } from '@shopgate/engage/core';
+import { router } from '@virtuous/conductor';
+import CategoryPicker from './components/Picker';
+import { useNestedCategoryFilterWidget } from './hooks';
+import WidgetHeadline from '../../components/WidgetHeadline';
+import { useLocalStorage } from '../../hooks';
+
+const { colors, variables } = themeConfig;
+
+const useStyles = makeStyles()({
+  container: {
+    background: colors.light,
+    display: 'flex',
+    flexDirection: 'column',
+    paddingBottom: variables.gap.big,
+  },
+  buttonContainer: {
+    padding: `0 ${variables.gap.big}px`,
+  },
+  button: {
+    width: '100%',
+  },
+});
+
+/**
+ * The NestedCategoryFilter component
+ * @returns {JSX.Element}
+ */
+const NestedCategoryFilter = () => {
+  const { classes } = useStyles();
+  const { state: routeState, id: routeId } = useRoute();
+  const {
+    maxDepth = 3,
+    category,
+    code,
+    level1Label,
+    level2Label,
+    level3Label,
+    showHeadline,
+    headline,
+    rememberSelection,
+  } = useNestedCategoryFilterWidget();
+
+  const prevCategory = usePrevious(category);
+
+  const LOCAL_STORAGE_KEY = `nestedCategoryFilterState-${code}`;
+  const [storedPickers, setStoredPickers] = useLocalStorage(LOCAL_STORAGE_KEY);
+
+  const initialPickers = rememberSelection && storedPickers
+    ? storedPickers : routeState[code]?.pickers
+    ?? [{
+      categoryId: category,
+      selectedId: null,
+    }];
+
+  const [pickers, setPickers] = useState(initialPickers);
+  const [buttonCategoryId, setButtonCategoryId] = useState(null);
+
+  useEffect(() => {
+    setStoredPickers(pickers);
```

Comment:
local storage is updated whenever pickers change - even when the `rememberSelection` setting is off.